Sources/PackageLoading/ManifestJSONParser.swift

```diff
@@ -315,213 +304,70 @@ enum ManifestJSONParser {
 
-        let sources: [String]? = try? json.get("sources")
-        try sources?.forEach{ _ = try RelativePath(validating: $0) }
+        try target.sources?.forEach{ _ = try RelativePath(validating: $0) }
+        try target.exclude.forEach{ _ = try RelativePath(validating: $0) }
 
-        let exclude: [String] = try json.get("exclude")
-        try exclude.forEach{ _ = try RelativePath(validating: $0) }
-
-        let pluginUsages = try? json
-            .getArray("pluginUsages")
-            .map(TargetDescription.PluginUsage.init(v4:))
+        let pluginUsages = target.pluginUsages?.map { TargetDescription.PluginUsage.init($0) }
 
         return try TargetDescription(
-            name: try json.get("name"),
+            name: target.name,
             dependencies: dependencies,
-            path: json.get("path"),
-            url: json.get("url"),
-            exclude: exclude,
-            sources: sources,
-            resources: try Self.parseResources(json),
-            publicHeadersPath: json.get("publicHeadersPath"),
-            type: try .init(v4: json.get("type")),
-            pkgConfig: json.get("pkgConfig"),
+            path: target.path,
+            url: target.url,
+            exclude: target.exclude,
+            sources: target.sources,
+            resources: try Self.parseResources(target.resources),
+            publicHeadersPath: target.publicHeadersPath,
+            type: .init(target.type),
+            pkgConfig: target.pkgConfig,
             providers: providers,
             pluginCapability: pluginCapability,
-            settings: try Self.parseBuildSettings(json),
-            checksum: json.get("checksum"),
+            settings: try Self.parseBuildSettings(target),
+            checksum: target.checksum,
             pluginUsages: pluginUsages
         )
     }
 
-    private static func parseResources(_ json: JSON) throws -> [TargetDescription.Resource] {
-        guard let resourcesJSON = try? json.getArray("resources") else { return [] }
-        return try resourcesJSON.map { json in
-            let rule = try json.get(String.self, forKey: "rule")
-            let path = try RelativePath(validating: json.get(String.self, forKey: "path"))
-            switch rule {
+    private static func parseResources(_ resources: [Serialization.Resource]?) throws -> [TargetDescription.Resource] {
+        return try resources?.map {
+            let path = try RelativePath(validating: $0.path)
+            switch $0.rule {
             case "process":
-                let localizationString = try? json.get(String.self, forKey: "localization")
-                let localization = localizationString.map({ TargetDescription.Resource.Localization(rawValue: $0)! })
+                let localization = $0.localization.map({ TargetDescription.Resource.Localization(rawValue: $0.rawValue)! })
```

Comment:
! safe here?

Reply:
Taken from the original code, so "yes" :)